flink-core/src/main/java/org/apache/flink/configuration/ConfigOptionsDocGenerator.java

```diff
@@ -0,0 +1,279 @@
+ * limitations under the License.
+ */
+package org.apache.flink.configuration;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.flink.api.java.tuple.Tuple2;
+
+/**
+ * Class used for generating code based documentation of configuration parameters.
+ */
+public class ConfigOptionsDocGenerator {
+
+	/**
+	 * Transforms this configuration group into HTML formatted table.
+	 * Options are sorted alphabetically by key.
+	 *
+	 * @param options list of options to include in this group
+	 * @return string containing HTML formatted table
+	 */
+	private static String toHTMLTable(final List<ConfigOption> options) {
+		final StringBuilder htmlTable = new StringBuilder(
+			"<table class=\"table table-bordered\"><thead><tr><th class=\"text-left\" style=\"width: 20%\">Key</th>" +
+			"<th class=\"text-left\" style=\"width: 15%\">Default Value</th><th class=\"text-left\" " +
+			"style=\"width: 65%\">Description</th></tr></thead><tbody>");
+
+		for (ConfigOption option : options) {
+			htmlTable.append(toHTMLString(option));
+		}
+
+		htmlTable.append("</tbody></table>");
+
+		return htmlTable.toString();
+	}
+
+	/**
+	 * Transforms this configuration group into HTML formatted table.
+	 * Options are sorted alphabetically by key.
+	 *
+	 * @param clazzz a class that contains options to be converted int documentation
+	 * @return string containing HTML formatted table
+	 */
+	static String create(Class<?> clazzz) {
+		final List<ConfigOption> configOptions = extractConfigOptions(clazzz);
+
+		return toHTMLTable(configOptions);
+	}
+
+	private static List<ConfigOption> extractConfigOptions(Class<?> clazzz) {
+		try {
+			final List<ConfigOption> configOptions = new ArrayList<>();
+			final Field[] fields = clazzz.getFields();
+			for (Field field : fields) {
+				if (field.getType().equals(ConfigOption.class) && field.getAnnotation(Deprecated.class) == null) {
+					configOptions.add((ConfigOption) field.get(null));
+				}
+			}
+
+			return configOptions;
+		} catch (Exception e) {
+			throw new RuntimeException("Could not retrieve all options.", e);
+		}
+	}
+
+	/**
+	 * Transforms option to table row.
+	 *
+	 * @param option option to transform
+	 * @return row with the option description
+	 */
+	private static String toHTMLString(final ConfigOption<?> option) {
+		return "<tr>" +
+				"<td>" + option.key() + "</td>" +
+				"<td>" + defaultValueToHtml(option.defaultValue()) + "</td>" +
+				"<td>" + option.description() + "</td>" +
+				"</tr>";
+	}
+
+	private static String defaultValueToHtml(Object value) {
```

Comment:
the naming is inconsistent in regards to whether "Html" is upper-case or not. I have no preference for either though.